### PR TITLE
Colonies: Introduce GameObjects

### PR DESCRIPTION
**Pull request by Jusctsch5**
- Adding Game Board, Parser, Logger, and Pieces.
- Add basic shell of game abstraction
- Add ability to parse input file and create GameBoard
- Create folder structure

### CODE REVIEW

**Comment by Jusctsch5** [comment]
This is my first merge request!

File: .gitignore [hunk 1]
```diff
@@ -5,6 +5,9 @@ __pycache__/
 *.py[cod]
 *$py.class
 
+# Special Colonies files
+*.log
+
```
Comment (Jusctsch5) [inline .gitignore:10]: It would be nice to ignore eclipse project files too.

File: .project [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (golmschenk) [inline .project:1]: This file should be ignored

**Comment by golmschenk** [comment]
The project files should be gitignored

File: source/gameBoard.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+class GameBoard:
+    """GameBoard describes the area where a game is played. """
+
+    numBoards = 0
+
+    def __init__(self, iWidth=0, iHeight=0):
+        """
+        Constructor for a GameBoard.
+        Args:
+            iWidth - The width of the board.
```
Comment (golmschenk) [inline source/gameBoard.py]: For reStructuredText

```
:param iWidth: The width of the board.
:type iWidth: int
```

No `Args:` line. Same indent as upper comment.

If iWidth could be multiple types:

```
:type iWidth: int or float or None
```

Finally return stuff:

```
:return: The thing being returned description.
:rtype: int
```

File: source/gamePiece.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+from enum import Enum
```
Comment (Jusctsch5) [inline source/gamePiece.py]: Remove the enumeration import and philosophical comments.

File: .project (hunk 2 above)
Comment (Jusctsch5) [inline .project:1]: Done

File: source/gameBoard.py (hunk 3 above)
Comment (Jusctsch5) [inline source/gameBoard.py]: Done.

File: source/gamePiece.py (hunk 4 above)
Comment (Jusctsch5) [inline source/gamePiece.py]: Done.

**Pull request merged**